data_model/src/lib.rs

```diff
@@ -741,6 +741,16 @@ pub trait Registered: Identifiable {
     type With: Into<RegistrableBox>;
 }
 
+/// Trait for proxy objects used for registration.
+#[cfg(feature = "mutable_api")]
+pub trait Registrable {
+    /// Constructed type
+    type Target: Identifiable + Registered<With = Self>;
```

Comment:
```suggestion
    type Target: Registered<With = Self>;
```

Seems that it's implied by `Registered`

Reply:
or should we remove the bound from `Registered`?

Reply:
I think `Registered` is unnecessary, so we can remove it.

Reply:
Actually we can omit any trait bounds on `Target`, since they are not used.

Reply:
Are you getting rid of `Registered`?

Reply:
Yes, in `Target` trait bounds.